app/src/main/java/com/eveningoutpost/dexdrip/Services/G5CollectionService.java

```diff
@@ -266,13 +266,11 @@ public void onDestroy() {
 
         Log.d(TAG, "onDestroy");
         SharedPreferences sharedPrefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
-        if (sharedPrefs.getBoolean("use_wear_connectG5", false)) {//KS
```

Comment:
Yes, this was a fix.  I forget now the exact problem but it was causing the Wear G5CollectionService to not shutdown properly.  Instead, I moved the shutdown criteria into ListenerService which has more control and logic to determine whether G5 should be shutdown.
A related fix in the app/CollectionServiceStarter.java line 182.